src/hacbs/utils/commits-utils.ts

```diff
@@ -62,3 +71,52 @@ export const getCommitsFromPLRs = (plrList: PipelineRunKind[], limit?: number):
   );
   return limit && limit < sortedCommits.length ? sortedCommits.slice(0, limit) : sortedCommits;
 };
+
+export const getCommitByName = (pipelineruns: PipelineRunKind[], commitName: string): Commit => {
+  const plr = pipelineruns.find(
+    (pipelinerun) => pipelinerun.metadata.labels[PipelineRunLabel.COMMIT_LABEL] === commitName,
+  );
+  if (!plr) {
+    return null;
+  }
+  const commitSHA = plr.metadata.labels?.[PipelineRunLabel.COMMIT_LABEL];
+  const commitBranch =
+    plr.metadata.annotations?.[PipelineRunLabel.COMMIT_BRANCH_ANNOTATION]
+      ?.slice(1, -1)
+      .split(',') ?? [];
```

Comment:
I don't believe that you will get a list of branches as the commit represents a single branch.
Please confirm with @karthikjeeyar and tekton.

Reply:
@abhinandan13jan I see that you are using `COMMIT_BRANCH_ANNOTATION = pipelinesascode.tekton.dev/on-target-branch` which is used by tekton internally to match an event (push/pull) against  a list of target branches, but As Christian mentioned a commit will happen only on a branch, you need to use this `build.appstudio.redhat.com/target_branch: {{target_branch}}` annotation to get the matched branch value.

 Refer this rules for pipeline customisation annotations section - https://docs.google.com/document/d/1PXkpFHKrnq1Sg1giTgeXdYzNVf7CTRgwowykr7YPM2I/edit#